### DISCUSSION TITLE
Run command from URL or similar?

### DISCUSSION THREAD

**Discussion by themonk30** [discordjs/discord.js#5607] [Q&A] [answered]
Can anyone advise please - I would like to run a bot command that is executed by a PHP Script....
So once they register on my website, they get added to the DB, they receive an email (all done in PHP) - but then I want to pass a command over to my bot (which is working) that will automatically add that user to a role.

They run a command in Discord to get started, so I store their User ID and their Discord UN/Tag in the DB. 

!register -> Stores details, send them an email to complete registration process
-> Go to website, complete registration, the get access to a membership area on website
-> At this point, I want to add a role to their discord on my server.

Can this be achieved?

Thank you

**Comment by kyranet** [comment, marked as answer]
Afaik nothing stops you from calling said API yourself, or porting that PHP script to Node.js for that matter.

I don't think this should be something the library should do, specially when there can be many factors, such as using different protocols (RPC, WS, [AMQP](https://en.wikipedia.org/wiki/Advanced_Message_Queuing_Protocol) such as RabbitMQ...) in either unencrypted, encrypted with TLS/SSL, or SSH-tunneled.

As you'd notice, this can be a very complicated thing, and would easily go outside of what the library should handle.

**Comment by themonk30** [comment]
I don't expect it's necessarily something the library should or shouldn't do, I guess the question is, could I execute, in some way - the NodeJS file (in PHP) with parameters to set a user ID with a role once PHP script does its thing.

**Comment by kyranet** [comment]
Yes, you can do that, since it's a PHP server, you just query to it using a HTTP library such as [`node-fetch`](https://www.npmjs.com/package/node-fetch).

Edit: we converted this from an issue to a draft.

**Comment by themonk30** [comment]
Appreciate the guidance, kyranet. I guess the alternative is call the Discord API directly with PHP?

**Reply by kyranet** [reply]
You could do that, but it's ill-advised as discord.js won't be able to handle your PHP server's ratelimits.

Instead, you query to it and then relay the data to discord.js so it may be handled correctly.
